pkg/patterns/declarative/reconciler.go

```diff
@@ -173,6 +190,43 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 		return reconcile.Result{}, fmt.Errorf("error applying manifest: %v", err)
 	}
 
+	statusMap := make(map[status.Status]bool)
+	for _, object := range objects.Items {
+
+		unstruct, err := getObjectFromCluster(object, r)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		res, err := status.Compute(unstruct)
+		if err != nil {
+			log.WithValues("unstuctured", unstruct).Error(err, "Unable to get status of unstructured")
+			statusMap[status.NotFoundStatus] = true
+		}
+		if res != nil {
+			log.WithValues("kind", object.Kind).WithValues("name", object.Name).WithValues("status", res.Status).WithValues(
+				"message", res.Message).Info("Got status of resource:")
+			statusMap[res.Status] = true
+		}
+	}
+
+	addonObject, ok := instance.(addonsv1alpha1.CommonObject)
```

Comment:
I've realized we should probably have some helper functions for this, so that we can deal with unstructured also!  We might also be able then to eliminate the need for the methods and just use reflection instead, though I'm not sure if that actually helps us that much.